### ISSUE TITLE
bild.de

### ISSUE THREAD

**Issue by hardy123480** [reek/anti-adblock-killer#664]
Hello guys, here's a big one. bild.de completely blocks the access to their site since this morning. They're saying it is a test to force people to disable Adblockers but the thing is, most people are not willing to support this site because of their sh't journalism. Most people only visit the site for their sport news.

**Comment by 3941** [comment]
Site: http://bild.de
Screenshot: http://i.imgur.com/X1jlVBO.png
Browser: Firefox 41.0.1
Scriptmanager: Greasemonkey
Adblockler: uBlock Origin + uMatrix for script blocking
Filterlists: http://i.imgur.com/mqwULti.png

The Anti-Adblock script seems to be hosted by code.bildstatic.de .
When I block those scripts I can read the articles, but videos and other dynamic parts of the site won't work.

**Comment by hardy123480** [comment]
Ok, got a solution:

Filter:
http://code.bildstatic.de/1.20.4-release--15.20/bild/built/Konfig/js/packages/core.js
http://code.bildstatic.de/1.20.4-release--15.20/bild/built/Konfig/js/packages/BTOSeite.js

**Comment by 3941** [comment]
Blocking that core.js is enough, but ,as I said, it leaves the site in a rather crippled state.

**Comment by reek** [comment]
HI,

that sounds a complicated case, I would look at it when I have time

**Comment by hardy123480** [comment]
Hi, thanks for looking into it. Of course, we can't expect anything but I don't think it should be labeled as "low priority" because this site is the most visited news site in Germany. Blocking adblockers in this way is a huge dial. Maybe other news sites will do the same in the future.

**Comment by matbech** [comment]
The bild.de block reached quite some popular news sites today (slashdot etc). Maybe you want to reconsider the priority for some extra publicity :-)

**Comment by makloda** [comment]
This adblock script on Bild.de got news linked on Bloomberg http://www.bloomberg.com/news/articles/2015-10-13/axel-springer-tells-web-readers-to-turn-off-ad-blockers-or-pay

Looks like the company is betting big time on this.

**Comment by therealshuuk** [comment]
Please fix that problem. Its a really huge visit site in germany.

**Comment by 3941** [comment]
Looks like they removed the Anti-Adblock function again.
Maybe it wasn't working out quite as expected.

**Comment by alexander255** [comment]
Nope, it's still there: http://imgur.com/vAQB7f0 :-)

**Comment by 3941** [comment]
Huh... well that's weird. It works for me.
And they did change something about the javascript.

http://code.bildstatic.de/1.20.4-release--15.20/bild/built/Konfig/js/packages/core.js

changed to:

http://code.bildstatic.de/1.20.5-release--15.20/bild/built/Konfig/js/packages/core.js

1.20.4 to 1.20.5

**Comment by therealshuuk** [comment]
its still there  :(

**Comment by therealshuuk** [comment]
How can i block it now?

**Comment by hardy123480** [comment]
This finally works:

@@||sascdn.com^$domain=bild.de
@@||smartadserver.com^$domain=bild.de

**Comment by therealshuuk** [comment]
great
i need to add this in adblock plus or in greasemonkey anti adblock killer?

**Comment by hardy123480** [comment]
I'm using uBlock and just added this to my filters. Pretty sure it also works with Adblock Plus. Just block these two things.

**Comment by petrossa** [comment]
i confirm ublock with hardy123480's lines works perfectly. Just for those who use ghostery remember to whitelist the site.

**Comment by therealshuuk** [comment]
i deinstall. adblock plus and use ublock now and it works..thank you very much :)

**Comment by hardy123480** [comment]
I can't see any advertising there. Did you enable "Easy list Germany"? Did you update the filter lists?

Anyway, maybe anyone could add these two lines to the Adblock-Killer so that we don't need them in uBlock?

**Comment by 3941** [comment]
Don't be so quick about that.
I've been blocking those from the beginning, and still got that adblock message on the site.

Whatever they changed 11 hours ago, has changed their anti-adblock script.
It might not be final.

**Comment by latinlover** [comment]
Just working for some clicks... Then I'm again redirected to -> http://www.bild.de/wa/ll/bild-de/unangemeldet-42925516.bild.html

**Comment by hardy123480** [comment]
Working like a charm for me. Must have to do with your settings. As I said, I have uBlock + Easy List + Easy List Germany + these two lines. That's it. Working absolutely fine.

**Comment by therealshuuk** [comment]
I forget easylist-germany in ublock..works now thank you :)

**Comment by yoshimo** [comment]
from https://adblockplus.org/forum/viewtopic.php?f=2&t=41068#p140124

the code responsible for the adblock detection is in the "de.bild.user" namespace:

```
    ...
            getAbData: function() {
                return i()
            }
    ...
    function i() {
            var a = de.bild.cmsKonfig;
            return s === !1 && (s = {
                elemHidden: f(),
                saDisabled: g(),
                apf: h()
            },
            s.ba = s.elemHidden || s.saDisabled,
            a && a.d && parseInt(a.d) > 2 && (s.ba = s.ba || s.apf === !0)),       s
        }
```


the f() and g() functions just check if some ad-related stuff is there and fills elemHidden and saDisabled with true or false.
s.ba (which stands for blockedads or something, i guess) then gets set to true if either of those are blocked.

so basically it would be enough to wrap the getAdData function onload with something like:

```
    de.bild.user.getAbData = function() {
       var data = i();
       data.ba = data.elemHidden = data.saDisabled = false;
       return data;
    }
```

**Comment by frumpymoons** [comment]
> Working like a charm for me. Must have to do with your settings. As I said, I have uBlock + Easy List + Easy List Germany + these two lines. That's it. Working absolutely fine.

This doesn't work for me.

**Comment by hardy123480** [comment]
Then you're doing it wrong.

**Comment by 103655** [comment]
Maybe i know what frumpymoons mean. If i save the two lines, the page works very well. If i change to a subdomain: http://tv.bild.de/ (Ads will be displayed) and then a click on the headerpicture (Bild), the adblock-page will be displayed.

**Comment by hardy123480** [comment]
Can't confirm. Also works for me. Also no adds at tv.bild.de

**Comment by frumpymoons** [comment]
I dont know what i am doing wrong though. Maybe because i am on chrome?
I have Easylist + Easylist Ger and those two lines in my filters.

**Comment by reek** [comment]
Hi all,

I posted a new version of AakScript with a temporary solution, because suddenly the videos are not displayed. I'll try to find a solution to this problem

If I add filters proposed by @barthramsay
https://github.com/reek/anti-adblock-killer/pull/687

See you
Reek

**Comment by Thubb** [comment]
The filter does not work properly. It removes the adblocker warning, but async loading of images is broken and clicking the headline links in the teaser does not switch to the proper image and article.

When I disable the AntiAdblock Script, the page works fine again.

**Comment by barthramsay** [comment]
@Thubb Both things work for me with AntiAdBlock. I rechecked also today: Async image loading working and headline links ok. Also if you look closely at the filter they unblock stuff for the site and do NOT block additional resources. Because there are no other complains I suggest it is something with your other filter files.

**Comment by 3941** [comment]
@barthramsay Just tried disabling all my other filter add-ons.
Javascript of bild.de is now basically disabled by AAK.
Only when I disable AAK, I can get videos and any other dynamic part of the site to work.

**Comment by hacker999** [comment]
// ==UserScript==
// @name         Bild Sperre umgehen
// @version      0.1
// @author       unknown
// @match        http://*www.bild.de/*
// @grant        none
// @run-at       document-start
// ==/UserScript==
 
var interval = setInterval(function() {
    if (window.de) {
                window.de.bild.abTesting = function(a) { return; }
                clearInterval(interval);
   
    }
}, 0);

**Comment by Thubb** [comment]
Page works fine when I disable the script...

**Comment by barthramsay** [comment]
strange, I rechecked with uBlock in Chrome, Safari and Firefox and have no issues at all. Everything working including videos.
Which blocker & browser are you using?
And where are you based? I have the gut feeling (but did not checked) the site behaves differently based on location.

**Comment by 3941** [comment]
My system info is in the second post.
But just AAK alone is crippling the site for me.
Located in Germany of course. I guess most people here are. It's a german website after all.

**Comment by barthramsay** [comment]
My code addition is doing not much differently than what is suggested here https://github.com/reek/anti-adblock-killer/issues/664#issuecomment-147953118 and is only unblocking two sites (and sure... they are ad related sites).
You are also script blocking with uMatrix which seems to have side effects (I only use uBlock). Maybe you show the output of your browser console while loading the site. Deactivate everything which is messing up with scripts except uBlock and I'm sure it loads without Adblock warning.

**Comment by 3941** [comment]
As I've said like 3 or 4 posts earlier, I've already disabled everything but AAK and it messes up the site.

Fun fact: When I run everything but AAK, I can use the site just fine. No Anti Adblock Page coming up at all. Everything working. With uMatrix and uBlock Origin enabled and filtering.

The only thing ruining the site for me right now is indeed AAK.

I'm just happy that I don't particularly care. It's not my go-to news source.
Actually I'm here for a silly friend who basically can't live without it.

Now he still can't use it. I think he's only using Adblock+.

**Comment by barthramsay** [comment]
It seems there are two sorts of problem.
1. I can reproduce the problem with ABP+ on Firefox. ABP+ (with Easylist+Easylist Germany) with AAK seems not to work (only 1-2 clicks) where as Firefox + uBlock with exact same configuration and lists works. So clearly uBlock is doing a better job there and/or ABP+ is not as hidden as uBlock. This makes me very curious.
2. My experience is different. I always see the Anti Adblock except I activate AAK. I cannot reproduce your problem :( because I see dynamic parts (videos and images loading after page loads). As I said the 3 lines here https://github.com/reek/anti-adblock-killer/blob/master/anti-adblock-killer-filters.txt#L1438 only unblocks and not blocking anything more. Something in your script blocking is already taking care of bild.de and is in conflict with AAK (are you using any bild.de specific anti adblock JS scripts?). Do you also see no Anti Adblock with all script blocking/modification disabled and only Easylist+Easylist Germany (I've read from somebody that this works for him)?

**Comment by frumpymoons** [comment]
Can someone explain to me why I cant access the site at all on chrome? even if I disable every installed extension and deleting the cache?

**Comment by reek** [comment]
@hacker999
Thank you for this patch, which works great. I will add to the next version of AakScript

**Comment by terrorist96** [comment]
https://adblockplus.org/blog/smells-like-censorship-big-brother

**Comment by reek** [comment]
Hi,<br><br>Anti-AdBlock Killer has been updated ! <a href="https://github.com/reek/anti-adblock-killer#changelog">Changelog</a><br><br>Step 1: <a href="http://tinyurl.com/ong3bfe">Update</a> : UserScript<br>Step 2: <a href="http://reek.github.io/anti-adblock-killer/#fiterlist">Update</a> : FilterList<br><br>Do not forget to donate, review, share, follow and add to favorites.<br><br>See you<br>Reek

https://github.com/reek/anti-adblock-killer/
https://twitter.com/antiadbkiller

**Comment by Thubb** [comment]
Bild.de was working fine with release 8.7. It is broken again in 8.8... Same issues again, dynamic pic loading and clicking on the teasers does not work.

**Comment by reek** [comment]
@Thubb 
Your config please.

I know but it's the only solution I have found that works on all browser Although it causes some bugs but I have no other solution for now

Meanwhile here is the link for v7.7
https://github.com/reek/anti-adblock-killer/blob/v8.7/anti-adblock-killer.user.js

**Comment by Thubb** [comment]
What exactly do you want to know?
I am using FF 42, Greasmonkey, ublock origin, ghostery.
